databricks/koala/namespace.py

```diff
@@ -123,3 +126,59 @@ def _to_datetime2(arg_year, arg_month, arg_day,
         errors=errors,
         format=format,
         infer_datetime_format=infer_datetime_format)
+
+
+@derived_from(pd)
+def get_dummies(data, prefix=None, prefix_sep='_', dummy_na=False, columns=None, sparse=False,
+                drop_first=False, dtype=None):
+    if sparse is not False:
+        raise NotImplementedError("get_dummies currently does not support sparse")
+
+    if isinstance(columns, string_types):
+        columns = [columns]
+    if dtype is None:
+        dtype = 'byte'
+
+    if isinstance(data, Column):
+        if prefix is not None:
+            prefix = [str(prefix)]
+        columns = [data.name]
+        df = data.to_dataframe()
+        remaining_columns = []
+    else:
+        if isinstance(prefix, string_types):
+            raise ValueError("get_dummies currently does not support prefix as string types")
+        df = data.copy()
+        if columns is None:
+            columns = [column for column in df.columns
+                       if not isinstance(data.schema[column].dataType, NumericType)]
```

Comment:
Do you know if this is what pandas does? I thought it would accept also numeric values. In fact, I would filter for now for whitelisted types (string and maybe a few others), because I do not think this would work well with, say structures, in Spark (I have not tried).

Reply:
Yes, I check the pandas behavior and it only accepts non-numeric values by default.
If we explicitly specify the `columns` argument, it accepts numeric values as well.

```py
>>> import pandas as pd
>>> pdf = pd.DataFrame({'a': [0,1,2,0], 'b': list('abcb')})
>>> pd.get_dummies(pdf)
   a  b_a  b_b  b_c
0  0    1    0    0
1  1    0    1    0
2  2    0    0    1
3  0    0    1    0
>>> pd.get_dummies(pdf, columns=['a'])
   b  a_0  a_1  a_2
0  a    1    0    0
1  b    0    1    0
2  c    0    0    1
3  b    1    0    0
>>> pd.get_dummies(pdf, columns=['a', 'b'])
   a_0  a_1  a_2  b_a  b_b  b_c
0    1    0    0    1    0    0
1    0    1    0    0    1    0
2    0    0    1    0    0    1
3    1    0    0    0    1    0
```

I'll use the whitelist for the acceptable types.